InvenTree/build/api.py

```diff
@@ -65,6 +66,21 @@ def filter_assigned_to_me(self, queryset, name, value):
 
         return queryset
 
+    assigned_to = rest_filters.NumberFilter(label='responsible', method='filter_responsible')
+
+    def filter_responsible(self, queryset, name, value):
+        """Filter by orders which are assigned to the specified owner."""
+        owners = []
+        owners.append(Owner.objects.get(pk=value))
```

Comment:
You should use `Owner.objects.filter()` - `get()` can raise an exception if the matching PK does not exist

```python
owners = list(Owner.objects.filter(pk=value))
if len(owners) > 0 and owners[0].label() == 'user':
    ...
```

Reply:
thanks for the tip!

But using `User.objects.get()` should be fine, as it is already verified to exist by the first owners query, right?
Or is it generally favorable to use `.filter()` here as well?

Reply:
>  as it is already verified to exist by the first owners query, right?

I'm not 100% sure about that - you should check by trying to filter with a bad ID. If it does do that check for you, it'd be fine to use get

Reply:
looking good: any bad ID is successfully discarded by the first `Owner.objects.filter(pk=value)` already, so I'm happy with a `get()` here...